libraries/node-core-library/src/Text.ts

```diff
@@ -178,4 +200,72 @@ export class Text {
   public static escapeRegExp(literal: string): string {
     return literal.replace(/[^A-Za-z0-9_]/g, '\\$&');
   }
+
+  /**
+   * Read lines from an iterable object that returns strings or buffers, and return a generator that
+   * produces the lines as strings. The lines will not include the newline characters.
+   *
+   * @param iterable - An iterable object that returns strings or buffers
+   * @param options - Options used when reading the lines from the provided iterable
+   */
+  public static async *readLinesFromIterableAsync(
+    iterable: AsyncIterable<string | Buffer>,
+    options: IReadLinesFromIterableOptions = {}
+  ): AsyncGenerator<string> {
+    const { encoding = Encoding.Utf8, skipEmptyLines = false } = options;
+    const state: IReadLinesFromIterableState = { remaining: '' };
+    for await (const chunk of iterable) {
+      yield* Text._readLinesFromChunk(chunk, encoding, skipEmptyLines, state);
+    }
+    const remaining: string = state.remaining;
+    if (remaining.length) {
+      yield remaining;
+    }
+  }
+
+  /**
+   * Read lines from an iterable object that returns strings or buffers, and return a generator that
+   * produces the lines as strings. The lines will not include the newline characters.
+   *
+   * @param iterable - An iterable object that returns strings or buffers
+   * @param options - Options used when reading the lines from the provided iterable
+   */
+  public static *readLinesFromIterable(
+    // eslint-disable-next-line @rushstack/no-new-null
+    iterable: Iterable<string | Buffer | null>,
+    options: IReadLinesFromIterableOptions = {}
+  ): Generator<string> {
+    const { encoding = Encoding.Utf8, skipEmptyLines = false } = options;
+    const state: IReadLinesFromIterableState = { remaining: '' };
+    for (const chunk of iterable) {
+      yield* Text._readLinesFromChunk(chunk, encoding, skipEmptyLines, state);
+    }
+    const remaining: string = state.remaining;
+    if (remaining.length) {
+      yield remaining;
+    }
+  }
+
+  private static *_readLinesFromChunk(
+    chunk: string | Buffer | null,
+    encoding: Encoding,
+    skipEmptyLines: boolean,
+    state: IReadLinesFromIterableState
+  ): Generator<string> {
+    if (!chunk) {
+      return;
+    }
+    const remaining: string =
+      state.remaining + (typeof chunk === 'string' ? chunk : chunk.toString(encoding));
```

Comment:
Will this occasionally return a buffer?

Reply:
The iterable ReadableStream is typed such that it _could_ return either a buffer or a string (it's `AsyncIterable<string | Buffer>`, not `AsyncIterable<string> | AsyncIterable<buffer>`, so I don't think we can depend on one or the other being returned. Given the main use case for the method is for handling streams, it's a frustrating check to have to make, but IMO it may be needed. At least it's rather low overhead.

Reply:
In practice, no, but TypeScript doesn't know that because `setEncoding` does not alter the declared type of the object.